.github/workflows/build_and_test_pr.yml

```diff
@@ -0,0 +1,195 @@
+
+    # BUILD AND TEST INTEGRATION FILES ON THE BASELINE BRANCH
+
+    - name: Baseline Checkout
+      if: ${{ matrix.run_regressions }}
+      uses: actions/checkout@v4
+      with:
+        ref: develop
+        path: baseline
+
+    - name: Baseline Create Build Directory
+      if: ${{ matrix.run_regressions }}
+      run: cmake -E make_directory ./baseline/build/
+
+    - name: Baseline Configure CMake
+      if: ${{ matrix.run_regressions }}
+      working-directory: ./baseline/build
+      run: >
+        cmake 
+        -G "${{ matrix.generator }}"
+        -DCMAKE_BUILD_TYPE:STRING=Release
+        -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=${{ matrix.macos_dev_target }}
+        -DLINK_WITH_PYTHON:BOOL=ON
+        -DPython_REQUIRED_VERSION:STRING=${{ steps.setup-python.outputs.python-version }}
+        -DPython_ROOT_DIR:PATH=$RUNNER_TOOL_CACHE/Python/${{ steps.setup-python.outputs.python-version }}/${{ matrix.python-arch }}/
+        -DBUILD_TESTING:BOOL=ON
+        -DBUILD_FORTRAN:BOOL=ON 
+        -DBUILD_PACKAGE:BOOL=OFF
+        -DDOCUMENTATION_BUILD:STRING=DoNotBuild
+        -DENABLE_OPENMP:BOOL=OFF
+        -DUSE_OpenMP:BOOL=OFF
+        ../
+
+    - name: Baseline Build
+      if: ${{ matrix.run_regressions }}
+      working-directory: ./baseline/build
+      # Building specific target list so that we don't waste time building the unit test binary
+      run: cmake --build . -j ${{ matrix.nproc }} --config Release --target ${{ matrix.targets }}
+
+    - name: Baseline Test
+      if: ${{ matrix.run_regressions }}
+      working-directory: ./baseline/build
+      run: ctest -C Release -E Basement -R integration -j 3  # TODO: Speed up basement so we don't have to skip it.
+
+    # BUILD AND TEST EVERYTHING ON THE CURRENT BRANCH
+
+    - name: Branch Checkout
+      uses: actions/checkout@v4
+      with:
+        path: branch
+
+    - name: Branch Create Build Directory
+      run: cmake -E make_directory ./branch/build/
+
+    - name: Branch Configure CMake
+      working-directory: ./branch/build
+      run: >
+        cmake 
+        -G "${{ matrix.generator }}"
+        -DCMAKE_BUILD_TYPE:STRING=Release
+        -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=${{ matrix.macos_dev_target }}
+        -DLINK_WITH_PYTHON:BOOL=ON
+        -DPython_REQUIRED_VERSION:STRING=${{ steps.setup-python.outputs.python-version }}
+        -DPython_ROOT_DIR:PATH=$RUNNER_TOOL_CACHE/Python/${{ steps.setup-python.outputs.python-version }}/${{ matrix.python-arch }}/
+        -DBUILD_TESTING:BOOL=ON
+        -DBUILD_FORTRAN:BOOL=ON
+        -DBUILD_PACKAGE:BOOL=OFF
+        -DDOCUMENTATION_BUILD:STRING=DoNotBuild
+        -DENABLE_OPENMP:BOOL=OFF
+        -DUSE_OpenMP:BOOL=OFF
+        ../
+
+    - name: Branch Build
+      id: branch_build
+      working-directory: ./branch/build
+      run: cmake --build . -j ${{ matrix.nproc }} --config Release
+
+    - name: Branch Test
+      working-directory: ./branch/build
+      run: ctest -C Release -E Basement -j 3
+
```

Comment:
Next, if we are doing regressions, we will install the regression tool python package and run our new script to process results.